src/overlays/actors/ovl_En_Ani/z_en_ani.c

```diff
@@ -28,62 +44,314 @@ const ActorInit En_Ani_InitVars = {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->blinkFunc = EnAni_WaitForEyeOpen;
+        this->actionFunc = EnAni_Suffering;
+        Animation_Change(&this->skelAnime, &gAniHoldingFootPainWrithingAnim, 1.0f, 0.0f,
+                         Animation_GetLastFrame(&gAniHoldingFootPainWrithingAnim), 0, 0.0f);
+    }
+}
+
+/**
+ * pause: waiting for landing animation and hammer sfx to end
+ */
+void EnAni_WaitingForFootPain(EnAni* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->actionFunc = EnAni_FallingOver;
+        Animation_Change(&this->skelAnime, &gAniFallOverHoldingFootAnim, 1.0f, 0.0f,
+                         Animation_GetLastFrame(&gAniFallOverHoldingFootAnim), 2, 0.0f);
+    }
+}
 
-extern ColliderCylinderInit D_80968670;
-extern InitChainEntry D_8096869C[];
+void EnAni_FallingToGround(EnAni* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s16 quakeValue;
 
-extern UNK_TYPE D_06000C14;
-extern UNK_TYPE D_060011CC;
-extern UNK_TYPE D_060027A0;
-extern UNK_TYPE D_06009220;
-extern UNK_TYPE D_06009D34;
+    // if hit the ground
+    if ((this->actor.bgCheckFlags & 1) != 0) {
+        this->actor.flags &= ~0x10;
+        this->actionFunc = EnAni_WaitingForFootPain;
+        this->actor.velocity.x = 0.0f;
+        this->actor.velocity.z = 0.0f;
+        // the animation gets cut short, (first 16 frames only) only the landing part is seen
+        Animation_Change(&this->skelAnime, &gAniLandingThenStandingUpAnim, 1.0f, 0.0f, 16.0f, 2, 0.0f);
+        this->stateFlags |= ANI_STATE_SUFFERING;
+        quakeValue = Quake_Add(globalCtx->cameraPtrs[0], 3);
+        Quake_SetSpeed(quakeValue, 0x6978);
+        Quake_SetQuakeValues(quakeValue, 7, 0, 0, 0);
+        Quake_SetCountdown(quakeValue, 0x14);
+        Audio_PlayActorSound2(&this->actor, NA_SE_IT_HAMMER_HIT);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_809679D0.s")
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 2, 2000, 0x100);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_80967A48.s")
+void EnAni_LosingBalance(EnAni* this, GlobalContext* globalCtx) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_80967AB4.s")
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->actor.minVelocityY = -20.0f;
+        this->actor.gravity = -5.0f;
+        this->actor.velocity.y = 0.0f;
+        this->actor.velocity.z = -4.0f;
+        // frame count : 0.0f, only first frame, rest is handled in next action func
+        Animation_Change(&this->skelAnime, &gAniLandingThenStandingUpAnim, 0.0f, 0.0f, 0.0f, 2, 5.0f);
+        this->actionFunc = EnAni_FallingToGround;
+        gSaveContext.eventInf[1] |= 0x10;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/EnAni_Init.s")
+void EnAni_TreeHanging(EnAni* this, GlobalContext* globalCtx) {
```

Comment:
General comment: we usually use the simple present rather than the participle/continuous present, so `HangInTree` here, I think.